### ISSUE TITLE
Problem with apt update / apt upgrade on Raspberry Pi

### ISSUE THREAD

**Issue by Falconcrest** [ntop/ntopng#6804]
Linux ntopng 5.15.32-v8+ #1538 SMP PREEMPT Thu Mar 31 19:40:39 BST 2022 aarch64 GNU/Linux
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
ntopng Professional v.5.5.220717
Raspberry Pi 3 Model B Plus Rev 1.3

When I run apt-get update I get the following error messages:

pi@ntopng:~ $ sudo apt-get update
Hit:1 http://deb.debian.org/debian bullseye InRelease
Hit:2 http://deb.debian.org/debian bullseye-updates InRelease
Hit:3 http://security.debian.org/debian-security bullseye-security InRelease
Get:6 http://archive.raspberrypi.org/debian bullseye InRelease [23.7 kB]
Get:4 https://packages.ntop.org/apt/bullseye_pi armhf/ InRelease [2,155 B]
Get:5 https://packages.ntop.org/apt/bullseye_pi all/ InRelease [2,153 B]
Get:7 http://archive.raspberrypi.org/debian bullseye/main arm64 Packages [286 kB]
Get:8 http://archive.raspberrypi.org/debian bullseye/main armhf Packages [295 kB]
**_### Err:4 https://packages.ntop.org/apt/bullseye_pi armhf/ InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
Err:5 https://packages.ntop.org/apt/bullseye_pi all/ InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599_**
Fetched 609 kB in 3s (181 kB/s)
Reading package lists... Done
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://packages.ntop.org/apt/bullseye_pi armhf/ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://packages.ntop.org/apt/bullseye_pi all/ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
W: Failed to fetch http://apt.ntop.org/bullseye_pi/armhf/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
W: Failed to fetch http://apt.ntop.org/bullseye_pi/all/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
W: Some index files failed to download. They have been ignored, or old ones used instead.

What can I do?

**Comment by cardigliano** [comment]
Please reinstall the apt-ntop package from https://packages.ntop.org as reported in this FAQ https://www.ntop.org/support/faq/howto-update-gpg-package-signing-keys/

**Comment by Falconcrest** [comment]
I should have read better... 😉
I first tried reinstalling following this guide... 

-------------------------------
Raspberry Raspbian Instructions
Type the following commands in your shell:

```
wget https://packages.ntop.org/RaspberryPI/apt-ntop_1.0.190416-469_all.deb sudo dpkg -i apt-ntop_1.0.190416-469_all.deb sudo apt-get update
sudo apt-get install ntopng nprobe n2n
```
------------------------------
...but that didn't work, gave the same errors.
Then I chose this path...

------------------------------
Debian 11 (Bullseye)

Before to install make sure to edit /etc/apt/sources.list and add "contrib" at the end of each line that begins with deb and deb-src. Then, type in your shell:

```
wget https://packages.ntop.org/apt-stable/bullseye/all/apt-ntop-stable.deb
apt install ./apt-ntop-stable.deb
```
-----------------------------

...and it worked.
There are no more errors with apt-get update etc.
Should that be a problem?
However, the version of ntopng has not changed, it is still v.5.5.220717. 
Is this correct for Raspian/Debian?

**Comment by cardigliano** [comment]
@Falconcrest I didn't realize you were using the RaspberryPI repo, let us check it

**Comment by cardigliano** [comment]
This seems to be a duplicate of #6795

**Comment by xvegax** [comment]
I have the same problem with my PI x64:

> $ sudo apt update
Hit:1 http://deb.debian.org/debian buster InRelease
Hit:2 http://deb.debian.org/debian-security buster/updates InRelease                                 
Get:3 http://deb.debian.org/debian buster-updates InRelease [56.6 kB]                                
Hit:4 https://download.docker.com/linux/debian buster InRelease                                      
Get:5 http://archive.raspberrypi.org/debian buster InRelease [32.6 kB]                               
Get:6 https://packages.ntop.org/apt/buster_pi armhf/ InRelease [2,155 B]                             
Get:7 https://packages.ntop.org/apt/buster_pi all/ InRelease [2,153 B]
Get:8 http://archive.raspberrypi.org/debian buster/main arm64 Packages [302 kB]
Get:9 http://archive.raspberrypi.org/debian buster/main armhf Packages [391 kB]
Err:6 https://packages.ntop.org/apt/buster_pi armhf/ InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
Err:7 https://packages.ntop.org/apt/buster_pi all/ InRelease                                                                                                                                       
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
Reading package lists... Done                                                                                                                                                                      
W: GPG error: https://packages.ntop.org/apt/buster_pi armhf/ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
E: The repository 'http://apt.ntop.org/buster_pi armhf/ InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: GPG error: https://packages.ntop.org/apt/buster_pi all/ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 3D84C955924F7599
E: The repository 'http://apt.ntop.org/buster_pi all/ InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.

I tried

`wget https://packages.ntop.org/apt-stable/buster/all/apt-ntop-stable.deb and
apt install ./apt-ntop-stable.deb`

but that did not solve the keys issue.
